sotodlib/io/hk_utils.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import yaml
+import itertools
+
+from so3g.hk import load_range
+from sotodlib import core
+
+
+def _get_swap_dict(d):
+    """
+    Swap dictionary keys and values; useful for organizing fields in
+    config file by device name instead of alias in get_grouped_hkdata().
+    """
+    return {v: k for k, v in d.items()}
+
+
+def get_grouped_hkdata(start, stop, config):
+    """
+    Takes output from load_range(), reconfigures load_range() dictionary
+    slightly. Groups HK field data by corresponding device/feed. Outputs
+    that list of data.
+    """
+    # call load_range()
+    print("running load_range()")
+    hkdata = load_range(start, stop, config=config)
+
+    # load all fields from config file
+    with open(config, 'r') as file:
+        hkconfig = yaml.safe_load(file)
+
+    field_dict = hkconfig['field_list']
+
+    # output a dict of field name, alias, timestamps, and data
+    # for fields that are online given start, stop and keys from load_range()
+    hk_aliases = list(hkdata.keys())
+
+    online_fields = {}
+    for alias in hk_aliases:
+        field_info = {alias: field_dict[alias]}
+        online_fields.update(field_info)
+
+    fields_data = {}
+    for alias in online_fields:
+        time = hkdata[alias][0]
+        data = hkdata[alias][1]
+        field = online_fields[alias]
+
+        info = {field: [alias, time, data]}
+        fields_data.update(info)
+
+    # now group field names by feed/device
+    sorted_fields = sorted(_get_swap_dict(online_fields))
+
+    grouped_feeds = []
+    key_func = lambda field: field.split('.')[0:4]
+    for key, group in itertools.groupby(sorted_fields, key_func):
+        grouped_feeds.append(list(group))
+
+    # use grouped_feeds and fields_data to output a grouped list of data per
+    # feed/device, ready to input to an axismanager
+    grouped_data = []
+    for group in grouped_feeds:
+        device_data = {}
+        for field in group:
+            field_dict = {field: fields_data[field]}
+            device_data.update(field_dict)
+
+        grouped_data.append(device_data)
+
+    return grouped_data
+
+
+def make_hkaman(grouped_data):
```

Comment:
I would have `make_hkaman` call `get_grouped_hkdata` (have it take in those parameters and call it inside the function). Users will pretty much always want to just use `make_hkaman` (or in that case call it `get_hkaman`?) I don't think there's a reason to make them call two functions each time. 

Or, if you want to keep those separate functions (because I can see how that's helpful) then make a `get_hkaman` the wraps the two. I just want the main entry point for users to be just one function.